### PR TITLE
Add show/hide field to schedule for jobs create modal

### PR DESCRIPTION
**Pull request by mlunoe**
---
~~⚠️ Related PR: https://github.com/mesosphere/reactjs-components/pull/329~~ Merged
~~⚠️ Updating reactjs-components needed for this to work properly https://github.com/dcos/dcos-ui/pull/960~~ Merged

---

### CODE REVIEW

**Comment by mlunoe** [comment]
Ping @daiyiwhale @wavesoft

**Comment by wavesoft** [comment]
Sorry, accidental UI click to the github desktop 😨  reverting...

**Comment by wavesoft** [comment]
I guess this is not in sync with #944 ? I had some troubles getting this to work with the new version of `reactjs-components`

**Comment by wavesoft** [comment]
Also `startingDeadlineSeconds` value is lost when I switch to JSON.

**Comment by mlunoe** [comment]
I will look into the `startingDeadlineSeconds`.

Not sure what you are referring to with this comment: https://github.com/dcos/dcos-ui/pull/948#issuecomment-239375890. This PR, as stated in the description, depended on a reactjs-components update, which landed. Did you do `npm i` before trying this out?

**Comment by mlunoe** [comment]
Arh... my bad @wavesoft. I forgotten to push the rebased commits

**Comment by mlunoe** [comment]
Good catch on the `startingDeadlineSeconds`! It should be transferred as expected now :)

File: src/js/components/JobForm.js [hunk 1]
```diff
@@ -8,18 +8,33 @@ const METHODS_TO_BIND = [
...
+    let scheduleEnabledChange = formData.hasOwnProperty('cron') &&
+      eventObj.eventType === 'change' &&
+      eventObj.fieldName === 'runOnSchedule';
+    if (eventObj.eventType !== 'blur' && !scheduleEnabledChange) {
+      return;
+    }
+
+    this.validateForm();
+    // Update model before we update multipleDefinition
+    this.model = this.triggerTabFormSubmit();
```
Comment (orlandohohmeier) [inline src/js/components/JobForm.js]: Would be nice if we find a way to not store the `model` on the form itself. But as this pattern is also found in the SchemaForm I'm ok if you don't address it.

**Comment by wavesoft** [comment]
Looks good to me!

**Comment by orlandohohmeier** [comment]
Needs a rebase

**Comment by mlunoe** [comment]
Ok, rebased. And also fixed a problem where the `forceUpdate` would cause focus to act weird changes in the labels not being transferred. The `forceUpdate` only happens when the enable schedule changes checkbox changes

**Pull request merged**